src/BeamAdapter/component/BeamInterpolation.inl

```diff
@@ -1210,18 +1210,19 @@ void BeamInterpolation<DataTypes>::InterpolateTransformUsingSpline(Transform& gl
                                                                    const Transform &global_H_local1,
                                                                    const Real &L)
 {
-    Vec3 P0,P1,P2,P3,dP01, dP12, dP03;
+    Vec3NoInit P0,P1,P2,P3;
 
     /// find the spline points
-     this->getControlPointsFromFrame(global_H_local0, global_H_local1,L,P0, P1,P2, P3);
+    this->getControlPointsFromFrame(global_H_local0, global_H_local1,L,P0, P1,P2, P3);
 
     Real bx=baryCoord;
-    Vec3 posResult;
+    Real invBx = 1-baryCoord;
+    Vec3NoInit posResult;
     Quat quatResult;
 
-    dP01 = P1-P0;
-    dP12 = P2-P1;
-    dP03 = P3-P0;
+    Vec3 dP01 = P1-P0;
+    Vec3 dP12 = P2-P1;
+    Vec3 dP03 = P3-P0;

```

Comment:
cannot do suggestion 🤫
- const all "precomputed" stuff (bx, invbx, etc)
- there is now a Quat(noinit) as well

Reply:
yes! 
In fact I added Quat(noinit) because of BA. but my changes has been lost somewhere.

Reply:
ah.. using QuatNoInit make this plugin not anymore compatible with SOFA 22.06